examples/win-proc-stat.c

```diff
@@ -0,0 +1,381 @@
+ * Structures and functions to store stat data
+ */
+typedef union pid_tid {
+    struct {
+        uint64_t pid : 32;
+        uint64_t tid : 32;
+    };
+    uint64_t value;
+} pid_tid_t;
+
+guint pid_tid_hash(gconstpointer key)
+{
+    pid_tid_t k = { .value = (uint64_t)key };
+    uint32_t hash = (k.pid & 0xffff) + (k.tid << 16);
+
+    return hash;
+};
+
+int pid_tid_cmp(const void* a, const void* b)
+{
+    pid_tid_t a_ = { .value = (uint64_t)a };
+    pid_tid_t b_ = { .value = (uint64_t)b };
+
+    bool rc = 0;
+    if (a_.pid == b_.pid && a_.tid == b_.tid)
+        rc = 1;
+
+    return rc;
+}
+
+void print_stats(gpointer key, gpointer value, __attribute__((unused)) gpointer userdata)
+{
+    pid_tid_t k = { .value = (uint64_t)key };
+    uint64_t v = (uint64_t)value;
+
+    printf("PID:%d,TID:%d,COUNT:%ld\n", k.pid, k.tid, v);
+}
+
+/*
+ * Map offset enums to actual structure+member or global variable/function names.
+ */
+enum win_offsets {
+    KPCR_PRCB,
+    KPCR_PRCBDATA,
+    KPRCB_CURRENTTHREAD,
+    ETHREAD_CID,
+    CLIENT_ID_UNIQUETHREAD,
+    __WIN_OFFSETS_MAX
+};
+
+static const char* win_offset_names[__WIN_OFFSETS_MAX][2] = {
+    [KPCR_PRCB] = {"_KPCR", "Prcb" },
+    [KPCR_PRCBDATA] = {"_KPCR", "PrcbData" },
+    [KPRCB_CURRENTTHREAD] = { "_KPRCB", "CurrentThread" },
+    [ETHREAD_CID] = {"_ETHREAD", "Cid" },
+    [CLIENT_ID_UNIQUETHREAD] = {"_CLIENT_ID", "UniqueThread" },
+};
+
+static bool profile_lookup_array(
+    vmi_instance_t vmi,
+    json_object* profile_json,
+    const char* symbol_subsymbol_array[][2],
+    addr_t array_size,
+    addr_t* rva)
+{
+    bool ret = false;
+
+    if (!profile_json) {
+        fprintf(stderr, "Rekall profile json is NULL!\n");
+        return ret;
+    }
+
+    int errors = 0;
+    for (size_t i = 0; i < array_size; i++) {
+        if (VMI_SUCCESS != vmi_get_struct_member_offset_from_json(
+                    vmi,
+                    profile_json,
+                    symbol_subsymbol_array[i][0],
+                    symbol_subsymbol_array[i][1],
+                    &rva[i])
+           ) {
+            errors++;
+            printf("Failed to find offset for %s:%s\n",
+                   symbol_subsymbol_array[i][0], symbol_subsymbol_array[i][1]);
+        }
+    }
+
+    if (errors == 0)
+        ret = true;
+
+    return ret;
+}
+
+static addr_t* fill_offsets_from_profile(vmi_instance_t vmi, const char* profile)
+{
+    addr_t* offsets = (addr_t*)g_malloc0(sizeof(addr_t) * __WIN_OFFSETS_MAX );
+    if ( !offsets )
+        return NULL;
+
+    json_object* profile_json = json_object_from_file(profile);
+    if (!profile_lookup_array(
+                vmi,
+                profile_json,
+                win_offset_names,
+                __WIN_OFFSETS_MAX,
+                offsets))
+        printf("Failed to find offsets for array of structure names and subsymbols.\n");
+
+    return offsets;
+}
+
+static int32_t get_current_thread_id(vmi_instance_t vmi, vmi_event_t* event)
```

Comment:
I'm afraid you copied a large chunk of this code verbatim from DRAKVUF. You can't do that, the license of DRAKVUF is incompatible with LibVMI's LGPL. I would suggest you close this PR and add this as an extra tool to DRAKVUF using the DRAKVUF license.